gcloud/logging/sink.py

```diff
@@ -28,16 +28,19 @@ class Sink(object):
 
     :type filter_: string
     :param filter_: the advanced logs filter expression defining the entries
-                    exported by the sink.
+                    exported by the sink.  If not passed, the instance should
+                    already exist, to be refreshed via :meth:`reload`.
 
     :type destination: string
     :param destination: destination URI for the entries exported by the sink.
+                        If not passed, the instance should already exist, to
+                        be refreshed via :meth:`reload`.
 
     :type client: :class:`gcloud.logging.client.Client`
     :param client: A client which holds credentials and project configuration
                    for the sink (which requires a project).
     """
-    def __init__(self, name, filter_, destination, client):
+    def __init__(self, name, filter_=None, destination=None, client=None):
```

Comment:
@daspecster `reload()` will do the Right Thing (TM) here:  it will fetch the resource JSON, and update the `filter_` and `destinatoin` properties from it.  `update()` and `create()` will fail, because `None` does not translate into a valid value for `filter_` (the default for `destination` is an empty string, which *is* valid).